opm/autodiff/GeoProps.hpp

```diff
@@ -304,6 +291,59 @@ namespace Opm
                              Opm::EclipseStateConstPtr eclState,
                              std::vector<double> &ntg);
 
+        template <class GridType>
+        void computePoreVolume_(const GridType &grid,
+                                Opm::EclipseStateConstPtr eclState)
+        {
+            int numCells = Opm::AutoDiffGrid::numCells(grid);
+            const int* globalCell = Opm::UgGridHelpers::globalCell(grid);
+            EclipseGridConstPtr eclGrid = eclState->getInputGrid();
+            const int nx = eclGrid->getNX();
+            const int ny = eclGrid->getNY();
+
+            // the "raw" pore volume.
+            const std::vector<double>& porvData =
+                eclState->get3DProperties().getDoubleGridProperty("PORV").getData();
+            pvol_.resize(numCells);
+
+            // the "activation number" grid property
+            const std::vector<int>& actnumData =
+                eclState->get3DProperties().getIntGridProperty("ACTNUM").getData();
+
+
+            for (int cellIdx = 0; cellIdx < numCells; ++cellIdx) {
+                const int cellCartIdx = globalCell[cellIdx];
+
+                double cellPoreVolume = porvData[cellCartIdx];
+
+                if (eclGrid->getMinpvMode() == MinpvMode::ModeEnum::OpmFIL) {
+                    // Sum the pore volumes of the cells above which have been deactivated
+                    // because their volume less is less than the MINPV threshold
+                    for (int aboveCellCartIdx = cellCartIdx - nx*ny;
+                         aboveCellCartIdx >= 0;
+                         aboveCellCartIdx -= nx*ny)
+                    {
+                        if (porvData[aboveCellCartIdx] >= eclGrid->getMinpvValue()) {
+                            // stop if we encounter a cell which has a pore volume which is
+                            // at least as large as the minimum one
+                            break;
+                        }
+
+                        const double aboveCellVolume = eclGrid->getCellVolume(aboveCellCartIdx);
+                        if (actnumData[aboveCellCartIdx] == 0 && aboveCellVolume > 1e-6) {
+                            // stop at explicitly disabled cells, but only if their volume is
+                            // greater than 10^-6 m^3
+                            break;
+                        }
```

Comment:
Actually, we do not have an alternative to OPMFIL today, so OPMFIL is basically the only thing we support. Still, this code seems to cater to a future context where we do support Eclipse's handling of MINPV+PINCH. Merging this today we will have equal pore volumes to Eclipse, which will make comparisons to Eclipse results look nicer. On the other hand, the cells will have a pore volume lower than what the final size of the cell implies. I think we can live with both, at least on a short term. On a longer term it tells me that we should clean up the whole pore volume calculation code, and if we can avoid it in opm-parser, move it to opm-grid.